File: test/basic/array.c
#include <stdio.h>
#include <stdlib.h>
#include "smack.h"

// @expect verified

#define MAXSIZE 10
#define RESET 0

int main() {
  int *a = (int*)malloc(MAXSIZE * sizeof(int));

  free(a);
  return 0;
}

File: test/rust/basic/array.rs
#[macro_use]
mod smack;
use smack::*;

// @expect verified

fn main() {
  let ar = [2u8.nondet(), 3u8.nondet(), 4u8.nondet()];
  let idx = 1usize.nondet();
  assume!(ar[0] < 4);
  assume!(ar[1] < 5);
  assume!(ar[2] < 6);
  assume!(idx < 3);
  assert!(ar[idx] <= 5);
}
